website/docs/Plugins/CorePlugins.md

```diff
@@ -0,0 +1,161 @@
+
+export class ClientHelloPlugin extends ClientPlugin {
+  static readonly id = 'hello-plugin';
+
+  onAgent(agent, sendToCore) {
+    agent.hello = async (name) => await sendToCore('hello-plugin', name));  
+  } 
+}
+
+export class CoreHelloPlugin extends CorePlugin {
+  static readonly id = 'hello-plugin';
+
+  onClientCommand({ puppetPage }, name) {
+    `Hello ${name}`);
+  } 
+}
+```
+
+As shown above, you can export multiple plugins from the same file. Also a client/core plugin combination can share the same `id` (unlike two core plugins, which must each have unique ids).
+
+To register this plugin in SecretAgent, just pass it to `agent.use()`. In the following example we pass through a path to the plugin file instead of the plugin class itself -- we're doing this because by default Core runs in a separate process from Client.  
+
+```javascript
+import agent from 'secret-agent';
+
+agent.use(require.resolve('./HelloPlugin'));
+
+await agent.hello('World');
+```
+
+The rest of this page documents the various functionalities you can add to your class.
+
+## Constructor
+
+### new CorePlugin<em>(createOptions)</em>
+New instance of CorePlugin is created for every agent instance. The createOptions object has three properties.
+
+#### **Arguments**:
+
+- createOptions `object` Receives the following:
+  - userAgentOption `UserAgentOption`. An object containing various attributes of the chosen userAgent (name, version, etc).
+  - browserEngine `BrowserEngine`. An instance containing the current BrowserEngine.
+  - plugins `Plugins`. An instance containing the current instance of Plugins attached to this session.
+  - logger `BoundLog`. An instance of logger you can use to log output.
+
+## Class Properties
+
+### CorePlugin.id *required*
+This should usually be set to the plugin's npm package name.
+#### **Type**: `string`
+
+### CorePlugin.type *required*
+This tells SecretAgent that the plugin is a CorePlugin. It must always be set.
+#### **Type**: `string`. This must always be set to `'CorePlugin'`.
+
+
+## Instance Method Hooks
+The following methods are optional. Add them to your plugin as needed.
+
+### configure<em>(config)</em>
+
+This hook is called during the initialization of a session/browserEmulator as well as every time agent.configure is called from the client. 
+
+#### **Arguments**:
+
+- config `object` Receives any (or none) of the following:
+  - viewport `Viewport`. This is an object containing browser width and height as well as screenWidth and screenHeight, among other properties.
+  - geolocation `Geolocation`. This is an object containing longtitude and latitude, among other properties.  
+  - timezoneId `string`. The configured unicode TimezoneId or host default (eg, America/New_York).
+  - locale `string`. The configured locale in use (eg, en-US).
+
+Modify any value in the object to change it session-wide.
+
+#### **Returns** `void`
+
+### onClientCommand<em>(meta, ...args)</em> *optional*
```

Comment:
Could use a client-side example here. It's hard to wrap your head around how this works/what you're supposed to do.

Reply:
The example at the top of the page is a onClientCommand example.